ergo-wallet/src/test/scala/org/ergoplatform/wallet/secrets/DerivationPathSpec.scala

```diff
@@ -19,4 +22,105 @@ class DerivationPathSpec
     }
   }
 
+  property("nextPath - new default derivation") {
+    implicit val enc = new ErgoAddressEncoder(ErgoAddressEncoder.MainnetNetworkPrefix)
+
+    // This testing pair is checked against CoinBarn and Yoroi
+    val mnemonic = "liar exercise solve delay betray sheriff method empower disease river recall vacuum"
+    val address = "9hAymcGaRfTX7bMADNdfWfk7CKzi2ZpvRBCmtEf6d92n8E26Ax7"
+
+    val seed = Mnemonic.toSeed(mnemonic, None)
+
+    val masterKey = ExtendedSecretKey.deriveMasterKey(seed)
+    val dp = DerivationPath.nextPath(IndexedSeq(masterKey)).get
+    val sk = masterKey.derive(dp).asInstanceOf[ExtendedSecretKey]
+    val pk = sk.publicKey.key
+
+    // The first derived key corresponds to m/44'/429'/0'/0/0 and the same as the key in CoinBarn
+    // (and the first key in Yoroi)
+    P2PKAddress(pk).toString() shouldBe address
+
+    val pk2 = masterKey.derive(DerivationPath.fromEncoded("m/44'/429'/0'/0/0").get).asInstanceOf[ExtendedSecretKey].publicKey.key
+    P2PKAddress(pk2).toString() shouldBe address
+
+    // next path should be m/44'/429'/0'/0/1
+    val dp2 = DerivationPath.nextPath(IndexedSeq(masterKey, sk)).get
+    dp2 shouldBe DerivationPath.fromEncoded("m/44'/429'/0'/0/1").get
+
+    // on top of old paths, derivation works as before EIP, m/1 -> m/2
+    val sk2 = masterKey.derive(DerivationPath.fromEncoded("m/1").get).asInstanceOf[ExtendedSecretKey]
+    val dp3 = DerivationPath.nextPath(IndexedSeq(masterKey, sk2)).get
+    dp3 shouldBe DerivationPath.fromEncoded("m/2").get
+
+    // for (m/1, m/44'/429'/0'/0/0), next path would be m/44'/429'/0'/0/1
+    val dp4 = DerivationPath.nextPath(IndexedSeq(masterKey, sk2, sk)).get
+    dp4 shouldBe DerivationPath.fromEncoded("m/44'/429'/0'/0/1").get
+  }
+
+
+  property("nextPath - old derivation") {
+    implicit val enc = new ErgoAddressEncoder(ErgoAddressEncoder.MainnetNetworkPrefix)
+
+    // pre-EIP3 derivation testing pair got from the node
+    val mnemonic = "liar exercise solve delay betray sheriff method empower disease river recall vacuum"
+    val address = "9h7f11AC9RMHkhFbXg46XfYHq3HNnb1A9UtMmMYo6hAuQzWxVWu"
+
+    val seed = Mnemonic.toSeed(mnemonic, None)
+
+    val masterKey = ExtendedSecretKey.deriveMasterKey(seed)
+    val dp = DerivationPath.nextPath(IndexedSeq(masterKey), usePreEip3Derivation = true).get
+    val sk = masterKey.derive(dp).asInstanceOf[ExtendedSecretKey]
+    val pk = sk.publicKey.key
+    P2PKAddress(pk).toString() shouldBe address
+
+    val dp2 = DerivationPath.nextPath(IndexedSeq(masterKey, sk)).get
```

Comment:
So, the usePreEip3Derivation = false is ignored in this case?

Reply:
Please describe the behavior for this test case in ScalaDoc of the method.

Reply:
@aslesarenko are we going to introduce ScalaDoc requirements for testing code ?

Reply:
@kushti no, I just don't understand the code. 
As always, please don't spend time writing he response here, improve the code instead, if you will.

Reply:
@aslesarenko ScalaDoc is not needed then it seems, added comments.